pyiceberg/catalog/hive.py

```diff
@@ -199,6 +184,7 @@ def _annotate_namespace(database: HiveDatabase, properties: Properties) -> HiveD
     DateType: "date",
     TimeType: "string",
     TimestampType: "timestamp",
+    TimestamptzType: "timestamp",
```

Comment:
In Java, we map `TimestamptzType` to `"timestamp with local time zone"` for Hive version >= 3 `TimestamptzType`. 
```java
case TIMESTAMP:
        Types.TimestampType timestampType = (Types.TimestampType) type;
        if (HiveVersion.min(HiveVersion.HIVE_3) && timestampType.shouldAdjustToUTC()) {
          return "timestamp with local time zone";
        }
        return "timestamp";
```
But in python it seems we do not have a way to check the version. So I put the "timestamp" here to ensure compatibility. Please correct me if I am wrong here. Thanks!


ref: https://github.com/apache/iceberg/blob/main/hive-metastore/src/main/java/org/apache/iceberg/hive/HiveSchemaUtil.java#L143

Reply:
Can we just set arbitrary strings in there? If so, I think `timestamp with local time zone` is more accurate. It would be good to validate using an integration test as well, since we have those tests already.

Reply:
Theoretically, the content here should be constants in [serdeConstants](https://github.com/apache/hive/blob/master/serde/src/gen/thrift/gen-javabean/org/apache/hadoop/hive/serde/serdeConstants.java#L76-L86) since java use these to [test the Hive schema util](https://github.com/apache/iceberg/blob/main/hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveSchemaUtil.java#L213)
Among these types, `timestamp with local time zone` is a new one added in Hive3: [ref](https://github.com/apache/iceberg/pull/1897).

Interestingly, when doing integration tests (use pyiceberg to write and spark to read) I found that Hive server will raise error for `timestampabc` but work normally for `timestamp arbitrary string`. Seems as long as the first word is within serdeConstants there will be no error loading the table. Not sure if `timestamp arbitrary string` will cause error in other Hive use cases.

How about we default to "timestamp with local time zone" and add a HiveCatalog property (e.g. `hive.hive2-compatible-mode`) to use "timestamp" here when it set to `true`?

(Will add the integration test soon)

Reply:
Thanks for the context, has been a while since working with the Hive catalog.

> How about we default to "timestamp with local time zone" and add a HiveCatalog property (e.g. hive.hive2-compatible-mode) to use "timestamp" here when it set to true?

I think that would be a great solution 👍